looks like your test also needs to remove an extra space character (the `received` is correct):
```
--- FAIL: TestRemoveSingleSigner (0.52s)
	Error Trace:	signer_remove_test.go:57
	Error:      	Error message not equal:
	            	expected: "No  signer test for image eiais/test2"
	            	received: "No signer test for image eiais/test2"
```

--- a/cli/command/trust/cmd.go
+++ b/cli/command/trust/cmd.go
@@ -19,6 +19,7 @@ func NewTrustCommand(dockerCli command.Cli) *cobra.Command {
 		newRevokeCommand(dockerCli),
 		newSignCommand(dockerCli),
 		newSignerAddCommand(dockerCli),
+		newSignerRemoveCommand(dockerCli),
 	)
 	return cmd
 }

--- a/cli/command/trust/helpers.go
+++ b/cli/command/trust/helpers.go
@@ -17,6 +17,7 @@ import (
 )
 
 const releasedRoleName = "Repo Admin"
+const releasesRoleTUFName = "targets/releases"
 
 func checkLocalImageExistence(ctx context.Context, cli command.Cli, imageName string) error {
 	_, _, err := cli.Client().ImageInspectWithRaw(ctx, imageName)

--- a/cli/command/trust/signer_remove.go
+++ b/cli/command/trust/signer_remove.go
@@ -1,0 +1,122 @@
+package trust
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/docker/cli/cli"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/trust"
+	"github.com/docker/notary/client"
+	"github.com/docker/notary/tuf/data"
+	"github.com/spf13/cobra"
+)
+
+type signerRemoveOptions struct {
+	forceYes bool
+}
+
+func newSignerRemoveCommand(dockerCli command.Cli) *cobra.Command {
+	options := signerRemoveOptions{}
+	cmd := &cobra.Command{
+		Use:   "signer-remove NAME IMAGE [IMAGE...]",
+		Short: "Remove a signer",
+		Args:  cli.RequiresMinArgs(2),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			return removeSigner(dockerCli, args[0], args[1:], &options)
+		},
+	}
+	flags := cmd.Flags()
+	flags.BoolVarP(&options.forceYes, "yes", "y", false, "Answer yes to removing most recent signer (no confirmation)")
+	return cmd
+}
+
+func removeSigner(cli command.Cli, signer string, images []string, options *signerRemoveOptions) error {
+	for _, image := range images {
+		if err := removeSingleSigner(cli, image, signer, options.forceYes); err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func isLastSignerForReleases(roleWithSig data.Role, allRoles []client.RoleWithSignatures) (bool, error) {
+	var releasesRoleWithSigs client.RoleWithSignatures
+	for _, role := range allRoles {
+		if role.Name == releasesRoleTUFName {
+			releasesRoleWithSigs = role
+			break
+		}
+	}
+	counter := len(releasesRoleWithSigs.Signatures)
+	if counter == 0 {
+		return false, fmt.Errorf("All signed tags are currently revoked, use docker trust sign to fix")
+	}
+	for _, signature := range releasesRoleWithSigs.Signatures {
+		for _, key := range roleWithSig.KeyIDs {
+			if signature.KeyID == key {
+				counter--
+			}
+		}
+	}
+	return counter < releasesRoleWithSigs.Threshold, nil
+}
+
+func removeSingleSigner(cli command.Cli, image, signerName string, forceYes bool) error {
+	_, ref, repoInfo, authConfig, err := getImageReferencesAndAuth(cli, image)
+	if err != nil {
+		return err
+	}
+	signerDelegation := data.RoleName("targets/" + signerName)
+	if signerDelegation == releasesRoleTUFName {
+		return fmt.Errorf("releases is a reserved keyword and cannot be removed")
+	}
+	notaryRepo, err := trust.GetNotaryRepository(cli, repoInfo, *authConfig, "push", "pull")
+	if err != nil {
+		return trust.NotaryError(ref.Name(), err)
+	}
+	delegationRoles, err := notaryRepo.GetDelegationRoles()
+	if err != nil {
+		return fmt.Errorf("Error retrieving signers for %s", image)
+	}
+	var role data.Role
+	for _, delRole := range delegationRoles {
+		if delRole.Name == signerDelegation {
+			role = delRole
+			break
+		}
+	}
+	if role.Name == "" {
+		return fmt.Errorf("No signer %s for image %s", signerName, image)
+	}
+	allRoles, err := notaryRepo.ListRoles()
+	if err != nil {
+		return err
+	}
+	if ok, err := isLastSignerForReleases(role, allRoles); ok && !forceYes {
+		removeSigner := command.PromptForConfirmation(os.Stdin, cli.Out(), fmt.Sprintf("The signer %s signed the last released version of %s. "+
+			"Removing them will lead to signed tags from %s being unpullable."+
+			"Are you sure you want to continue?: ",
+			signerName, image, image,
+		))
+
+		if !removeSigner {
+			fmt.Fprintf(cli.Out(), "\nAborting action.\n")
+			return nil
+		}
+	} else if err != nil {
+		fmt.Fprintf(cli.Out(), err.Error())
+	}
+	if err = notaryRepo.RemoveDelegationKeys(releasesRoleTUFName, role.KeyIDs); err != nil {
+		return err
+	}
+	if err = notaryRepo.RemoveDelegationRole(signerDelegation); err != nil {
+		return err
+	}
+	if err = notaryRepo.Publish(); err != nil {
+		return err
+	}
+	fmt.Fprintf(cli.Out(), "Successfully removed %s from %s\n", signerName, image)
+	return nil
+}

--- a/cli/command/trust/signer_remove_test.go
+++ b/cli/command/trust/signer_remove_test.go
@@ -1,0 +1,87 @@
+package trust
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"github.com/docker/cli/cli/internal/test"
+	"github.com/docker/docker/pkg/testutil"
+	"github.com/docker/notary/client"
+	"github.com/docker/notary/tuf/data"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTrustSignerRemoveErrors(t *testing.T) {
+	testCases := []struct {
+		name          string
+		args          []string
+		expectedError string
+	}{
+		{
+			name:          "not-enough-args-0",
+			expectedError: "requires at least 2 arguments",
+		},
+		{
+			name:          "not-enough-args-1",
+			args:          []string{"user"},
+			expectedError: "requires at least 2 arguments",
+		},
+		{
+			name:          "not-an-image",
+			args:          []string{"user", "notanimage"},
+			expectedError: "Error retrieving signers for notanimage",
+		},
+		{
+			name:          "sha-reference",
+			args:          []string{"user", "870d292919d01a0af7e7f056271dc78792c05f55f49b9b9012b6d89725bd9abd"},
+			expectedError: "invalid repository name",
+		},
+		{
+			name:          "invalid-img-reference",
+			args:          []string{"user", "ALPINE"},
+			expectedError: "invalid reference format",
+		},
+	}
+	for _, tc := range testCases {
+		cmd := newSignerRemoveCommand(
+			test.NewFakeCli(&fakeClient{}))
+		cmd.SetArgs(tc.args)
+		cmd.SetOutput(ioutil.Discard)
+		testutil.ErrorContains(t, cmd.Execute(), tc.expectedError)
+	}
+}
+
+func TestRemoveSingleSigner(t *testing.T) {
+	cli := test.NewFakeCli(&fakeClient{})
+	err := removeSingleSigner(cli, "eiais/test2", "test", true)
+	assert.EqualError(t, err, "No signer test for image eiais/test2")
+	err = removeSingleSigner(cli, "eiais/test2", "releases", true)
+	assert.EqualError(t, err, "releases is a reserved keyword and cannot be removed")
+}
+
+func TestIsLastSignerForReleases(t *testing.T) {
+	role := data.Role{}
+	releaserole := client.RoleWithSignatures{}
+	releaserole.Name = releasesRoleTUFName
+	releaserole.Threshold = 1
+	allrole := []client.RoleWithSignatures{releaserole}
+	lastsigner, _ := isLastSignerForReleases(role, allrole)
+	assert.Equal(t, false, lastsigner)
+
+	role.KeyIDs = []string{"deadbeef"}
+	sig := data.Signature{}
+	sig.KeyID = "deadbeef"
+	releaserole.Signatures = []data.Signature{sig}
+	releaserole.Threshold = 1
+	allrole = []client.RoleWithSignatures{releaserole}
+	lastsigner, _ = isLastSignerForReleases(role, allrole)
+	assert.Equal(t, true, lastsigner)
+
+	sig.KeyID = "8badf00d"
+	releaserole.Signatures = []data.Signature{sig}
+	releaserole.Threshold = 1
+	allrole = []client.RoleWithSignatures{releaserole}
+	lastsigner, _ = isLastSignerForReleases(role, allrole)
+	assert.Equal(t, false, lastsigner)
+
+}

